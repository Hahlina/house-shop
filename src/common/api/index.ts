export * from './fetchApi';
export * from './services/propety';
