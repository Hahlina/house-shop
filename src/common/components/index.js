export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
