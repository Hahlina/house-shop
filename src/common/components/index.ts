export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
export * from './Container/Container';
export * from './Image/Image';
export * from './ErrorPage/ErrorPage';
export * from './SomethingWentWrong/SomethingWentWrong';
export * from './NothingFound/NothingFound';
