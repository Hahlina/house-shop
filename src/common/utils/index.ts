export * from './convertPrice';
export * from './capitalizeFirstLetter';
