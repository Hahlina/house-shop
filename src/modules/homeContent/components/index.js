export * from "./PreviewBanner/PreviewBanner";
export * from "./PropertiesList/PropertiesList";
