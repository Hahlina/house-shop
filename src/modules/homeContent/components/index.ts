export * from './PreviewBanner/PreviewBanner';
export * from './PropertiesList/PropertiesList';
export * from './PropertiesPagination/PropertiesPagination';
