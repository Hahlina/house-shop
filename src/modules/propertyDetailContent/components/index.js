export * from "./PropertyGallery/PropertyGallery";
export * from "./FullInformationCard/FullInformationCard";
