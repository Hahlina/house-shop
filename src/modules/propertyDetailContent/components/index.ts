export * from './PropertyGallery/PropertyGallery';
export * from './PropertyGallery/PropertyGalleryLoader';
export * from './FullInformationCard/FullInformationCard';
export * from './FullInformationCard/FullInformationCardLoader';
export * from './PropertyDetailContentLoader/PropertyDetailContentLoader';
export * from './PropertyDetailBreadCrumbs/PropertyDetailBreadCrumbs';
