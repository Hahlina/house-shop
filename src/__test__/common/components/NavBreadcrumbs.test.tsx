import { NavBreadcrumbs } from 'common/components';

import { render, screen, mockedCustomLinkProps, mockedCustomPathNames } from 'common/utils/test';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: () => ({ pathname: '/example/path' })
}));

jest.mock('common/components/NavBreadcrumbs/NavBreadcrumbs.styled', () => ({
    LinkStyled: 'a'
}));

describe('NavBreadcrumbs component', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('should render component with default props without crashing', () => {
        render(<NavBreadcrumbs />);
        expect(screen.getByLabelText('breadcrumbs')).toBeInTheDocument();
    });

    it('should render component with customPathNames prop without crashing', () => {
        render(<NavBreadcrumbs customPathNames={mockedCustomPathNames} />);
        expect(screen.getByLabelText('breadcrumbs')).toBeInTheDocument();
    });

    it('should render component with customPathNames prop and custom LinkProps without crashing', () => {
        const customPathNames = mockedCustomPathNames.map((path) => ({ ...path, ...mockedCustomLinkProps }));
        render(<NavBreadcrumbs customPathNames={customPathNames} />);
        expect(screen.getByLabelText('breadcrumbs')).toBeInTheDocument();
    });

    it('should match snapshot with default props', () => {
        render(<NavBreadcrumbs />);
        expect(screen.getByLabelText('breadcrumbs')).toMatchSnapshot();
    });

    it('should match snapshot with customPathNames', () => {
        render(<NavBreadcrumbs customPathNames={mockedCustomPathNames} />);
        expect(screen.getByLabelText('breadcrumbs')).toMatchSnapshot();
    });
});
