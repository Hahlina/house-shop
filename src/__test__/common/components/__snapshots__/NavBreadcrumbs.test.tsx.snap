// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBreadcrumbs component should match snapshot with customPathNames 1`] = `
<nav
  aria-label="breadcrumbs"
  class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
>
  <ol
    class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
  >
    <li
      class="MuiBreadcrumbs-li"
    >
      <a
        to="/"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-qscxtw-MuiTypography-root"
        >
          Home
        </p>
      </a>
    </li>
    <li
      aria-hidden="true"
      class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
    >
      /
    </li>
    <li
      class="MuiBreadcrumbs-li"
    >
      <a
        to="/page1"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Page 1
        </p>
      </a>
    </li>
    <li
      aria-hidden="true"
      class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
    >
      /
    </li>
    <li
      class="MuiBreadcrumbs-li"
    >
      <a
        to="/page1/page2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Page 2
        </p>
      </a>
    </li>
  </ol>
</nav>
`;

exports[`NavBreadcrumbs component should match snapshot with default props 1`] = `
<nav
  aria-label="breadcrumbs"
  class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
>
  <ol
    class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
  >
    <li
      class="MuiBreadcrumbs-li"
    >
      <a
        to="/"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-qscxtw-MuiTypography-root"
        >
          Home
        </p>
      </a>
    </li>
    <li
      aria-hidden="true"
      class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
    >
      /
    </li>
    <li
      class="MuiBreadcrumbs-li"
    >
      <a
        to="/example"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Example
        </p>
      </a>
    </li>
    <li
      aria-hidden="true"
      class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
    >
      /
    </li>
    <li
      class="MuiBreadcrumbs-li"
    >
      <a
        to="/example/path"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Path
        </p>
      </a>
    </li>
  </ol>
</nav>
`;
